lib/src/repository/devices/devices_repository.dart

```diff
@@ -244,34 +252,41 @@ abstract class Property<T> {
 
 class SupportsDiscreteOnOff extends Property<bool> {
   SupportsDiscreteOnOff(bool value) : super('supportsDiscreteOnOff', value);
+  // Thoughts on below?
+  // super((SupportsDiscreteOnOff).toString(), value);
 }
 
 class SupportsIsJammed extends Property<bool> {
   SupportsIsJammed(bool value) : super('supportsIsJammed', value);
 }
 
 class LockTrait extends Trait {
+  // For backwards-compatibility
   LockTrait(State state, List<Property> properties)
+      : super('lock', [state], properties);
+
+  // Constructor receiving multiple states
+  LockTrait.multipleStates(List<State> state, List<Property> properties)
       : super('lock', state, properties);
 }
 
 class PowerTrait extends Trait {
   PowerTrait(State state, List<Property> properties)
```

Comment:
This was something that I came up with a question myself when doing this. Should it be mo betta if we just skipped this second argument and add a field on the trait for each property? Then we would have a constructor that takes the property values. Every trait seems to have distinctly unique properties. Then I'm not even sure we need a property class type.